Dockerfile

```diff
@@ -1,13 +1,13 @@
-ARG PHP_VERSION=7.4
+ARG PHP_VERSION=8.3
 
-FROM composer:2.1.12 as composer
+FROM composer:2.7.1 as composer
 
-FROM php:${PHP_VERSION}-cli-bullseye
+FROM php:${PHP_VERSION}-cli-bookworm
 
 COPY --from=composer /usr/bin/composer /usr/local/bin/composer
 
 RUN apt-get update && apt-get install -y --no-install-recommends \
-        git=1:2.30.2-1 libzip-dev=1.7.3-1 unzip=6.0-26 zip=3.0-12 \
+        git=1:2.39.* libzip-dev=1.7.* unzip=6.0* zip=3.0* \
```

Comment:
Do we need to be that prescriptive?

Reply:
Do you mean, "Do we need to be _that_ prescriptive?" or "Do we need to be prescriptive at all?" In answer to the former, no, but it's not particularly prescriptive, considering we specify `bookworm`. In fact, the `*zip*` versions cover the previous and next Debian releases, too. In answer to the latter question, yes, or else we turn off the Codacy check. As long as the Debian release matches what we have specified, it should no longer be an issue.

Reply:
I'd say more the latter. As the versions are essentially pinned by the distro, I don't think we should pin at all unless we have a region.

Reply:
I think it's more to do with the Docker build cache. If you don't pin them, and you change the `FROM` line but not the `apt-get` line, I believe the Docker build cache can sometimes load with incorrect versions. Then you get an error that's hard to diagnose.

Reply:
Ok, that's unexpected. Let's keep it as is, then.